src/Agent.Plugins/Artifact/ArtifactItemFilters.cs

```diff
@@ -231,6 +256,18 @@ private List<string> FilterItemsByPatterns(List<string> paths, IEnumerable<Func<
         }
 
         /// <summary>
+        /// Trims source path from full path of file. Result is path to file in artifact.
+        /// E.g. file is \\FileShare\TestArtifact\TestFolder\TestFile.txt, sourcePath is \\FileShare\TestArtifact, result is TestFolder\TestFile.txt.
+        /// </summary>
+        /// <param name="file">FileInfo object with info about file in artifact</param>
+        /// <param name="sourcePath">String path to artifact on file share.</param>
+        /// <returns></returns>
+        public string RemoveSourceDirFromPath(FileInfo file, string sourcePath)
+        {
+            char[] trimChars = new[] { '\\', '/' };
+            return file.ToString().Remove(0, sourcePath.Length).TrimStart(trimChars);
```

Comment:
Nit: could it make sense to add null checks?

Reply:
I think null-checks aren't required here.
File is item from list of these items (files). And it's caught from real dir on file share by `new DirectoryInfo(sourcePath).EnumerateFiles("*", SearchOption.AllDirectories).ToList<FileInfo>()` in method DownloadFileShareArtifactAsync. So only list of items can be null, but not FileInfo object in the list, IMHO.
sourcePath is string created by combining of 2 strings from buildArtifact properties. And I'm sure that sourcePath can't be null.